common/determined_common/storage/s3.py

```diff
@@ -32,6 +33,17 @@ def __init__(
             aws_secret_access_key=secret_key,
         )
 
+        # Detect if we are talking to minio, because boto3 has a client-side bug parsing the output
+        # of the minio server.
+        self._use_minio_workaround = False
+        if endpoint_url is not None:
+            try:
+                r = requests.get(endpoint_url)
+            except ConnectionError:
+                pass
+            else:
+                self._use_minio_workaround = r.headers.get("Server", "").lower() == "minio"
```

Comment:
non-blocking: Maybe add a log message here; seems like a good idea in general for any sort of protocol feature flag things like this.

Reply:
done.